packages/components/src/Popover/Popover.tsx

```diff
@@ -55,14 +57,48 @@ export interface PopoverProps extends UsePopoverProps {
    * The element which hovering on/off of will show/hide the triggering element
    */
   hoverDisclosureRef?: HTMLElement | null | RefObject<HTMLElement>
+  /**
+   * Convert popover to full-screen dialog on mobile
+   * @default false
+   */
+  renderMobileDialog?: boolean
+  /**
+   * Populate dialog header content when renderMobileDialog is true
+   */
+  dialogHeader?: string
```

Comment:
I think "label" might be more appropriate here.

```suggestion
  label?: string
```

Reply:
I'm just not sure if making that prop name more generic would cause confusion, considering 99.9% of popovers wouldn't even use it?